pod/video/views.py

```diff
@@ -2263,6 +2247,31 @@ def get_response_data(self, chunked_upload, request):
 @csrf_protect
 @login_required(redirect_field_name='referrer')
 def video_record(request):
+    if (RESTRICT_EDIT_VIDEO_ACCESS_TO_STAFF_ONLY
+            and request.user.is_staff is False):
+        return render(request,
+                      'videos/video_edit.html',
+                      {'access_not_allowed': True}
+                      )
+    if request.method == 'POST' and request.is_ajax():
+        try:
+            vid = Video()
+            vid.video = request.FILES['video']
+            vid.title = request.POST['title']
+            vid.owner = request.user
+            vid.type = Type.objects.get(id=DEFAULT_RECORDER_TYPE_ID)
+            vid.save()
+            vid.sites.add(get_current_site(request))
+            vid.launch_encode = True
+            vid.save()
```

Comment:
@ptitloup Juste une remarque ici, on est obligé de save avant d'ajouter le site et d'encoder ? Sinon c'est bon pour moi niveau code :+1:

Reply:
oui, tout à fait, il faut un id pour une relation many to many, du coup, je sauvegarde avant !